### ISSUE TITLE
can't register domain

### ISSUE THREAD

**Issue by so87** [evertramos/docker-compose-letsencrypt-nginx-proxy-companion#90]
# Error description:
This isn't an error on your alls part - I'm not sure how to get the certificate request working.  I can externally browse to the nginx container which gives me a "503" message with the version of nginx.  However, it still says the certificate is self signed.  My domain is "simonowens.crabdance.com" which is linked to my dynamic ip I get, so I don't have a static ip.  So for the variable "EXTERNAL_IP" I couldn't put "simonowens.crabdance.com" or any IP because I don't have a static one, so I just put the IP address the docker container is running on.  So do I need to have a static IP or is a dynamic one ok?

# Second Question
Nginx is smart enough to tell "hey this request came from simonowens.crabdance.com".   So when I ran the test container with "docker run -dit -e VIRTUAL_HOST=simonowens.crabdance.com --network=docker_sonarnet --name test-web httpd:alpine" i was never redirected to the apache page, still just the nginx 503 page.  This "docker_sonarnet" is just a bridged network, just given a virtual name.  Is this a problem that everything is bridged or do I need to only have these 3 docker containers bridged and all other containers I want to run on a virtual docker network?

Thank you for your time.  Sorry this wasn't really an issue and for the messy reporting back to you all.

**Comment by evertramos** [comment]
Hello @so87,

Thanks for your message. I am not quiet sure if I understood all your questions..

But regarding the test script and network here it is:

#### The proxy will create a bridge network for all containers including the proxy and the services containers you will run in your host. 

So, if you renamed your network in your *.env* file to *docker_sonarnet* and you are not sepparating the service network from your proxy network, all three containers (nginx-web/gen/letsencrypt) will be on the same network and your service container (test-web) as well, will be in this network.

As soon as you start your test it should show you the apache **it works** page!

So, if is not showing... your dns on the host *simonowens.crabdance.com* is probably not set it to the right IP address of your server.

About the first questiona... there is not a variable called EXTERNAL_IP.... there is an IP address which you could leave 0.0.0.0 to test it.. but this must have the server (host) IP address that communicate with external network. Normally is a public IP address but in some cases like Amazon AWS is a private ip address that redirects to the internet.

Please test it again, if possible start over, read carefully the readme and let me know if it worked.

**Comment by so87** [comment]
evertramos,

You did a great job of answering my questions and providing info.  I was able to leave the IP to 0.0.0.0 and have my router/firewall forward all of my traffic to the docker host these containers were running on.  Then i was able to make domains for myself, add them to corresponding docker containers, and then everything just worked!  

I did have to make all containers use the "webproxy" network in order to work.  I tried using the "multiple-networks" setup but I couldn't get it working right.  I'm pretty happy with how everything works now though.  Thank you for the quick response and feedback.
